graphql/client/types.go

```diff
@@ -0,0 +1,288 @@
+	// The kind of event (e.g. "ERC20TransferEvent").
+	Kind string `json:"kind"`
+	// The parameters for the event. The parameters are different for each event kind, but will always
+	// be a set of key-value pairs.
+	Parameters interface{} `json:"parameters"`
+}
+
+// The block number and block hash for the latest block that has been processed by Mesh.
+type LatestBlock struct {
+	Number *big.Int    `json:"number"`
+	Hash   common.Hash `json:"hash"`
+}
+
+// A signed 0x order according to the [protocol specification](https://github.com/0xProject/0x-protocol-specification/blob/master/v3/v3-specification.md#order-message-format).
+type NewOrder struct {
+	ChainID               *big.Int       `json:"chainId"`
+	ExchangeAddress       common.Address `json:"exchangeAddress"`
+	MakerAddress          common.Address `json:"makerAddress"`
+	MakerAssetData        []byte         `json:"makerAssetData"`
+	MakerAssetAmount      *big.Int       `json:"makerAssetAmount"`
+	MakerFeeAssetData     []byte         `json:"makerFeeAssetData"`
+	MakerFee              *big.Int       `json:"makerFee"`
+	TakerAddress          common.Address `json:"takerAddress"`
+	TakerAssetData        []byte         `json:"takerAssetData"`
+	TakerAssetAmount      *big.Int       `json:"takerAssetAmount"`
+	TakerFeeAssetData     []byte         `json:"takerFeeAssetData"`
+	TakerFee              *big.Int       `json:"takerFee"`
+	SenderAddress         common.Address `json:"senderAddress"`
+	FeeRecipientAddress   common.Address `json:"feeRecipientAddress"`
+	ExpirationTimeSeconds *big.Int       `json:"expirationTimeSeconds"`
+	Salt                  *big.Int       `json:"salt"`
+	Signature             []byte         `json:"signature"`
+}
+
+// A signed 0x order according to the [protocol specification](https://github.com/0xProject/0x-protocol-specification/blob/master/v3/v3-specification.md#order-message-format.)
+type Order struct {
+	ChainID               *big.Int       `json:"chainId"`
+	ExchangeAddress       common.Address `json:"exchangeAddress"`
+	MakerAddress          common.Address `json:"makerAddress"`
+	MakerAssetData        []byte         `json:"makerAssetData"`
+	MakerAssetAmount      *big.Int       `json:"makerAssetAmount"`
+	MakerFeeAssetData     []byte         `json:"makerFeeAssetData"`
+	MakerFee              *big.Int       `json:"makerFee"`
+	TakerAddress          common.Address `json:"takerAddress"`
+	TakerAssetData        []byte         `json:"takerAssetData"`
+	TakerAssetAmount      *big.Int       `json:"takerAssetAmount"`
+	TakerFeeAssetData     []byte         `json:"takerFeeAssetData"`
+	TakerFee              *big.Int       `json:"takerFee"`
+	SenderAddress         common.Address `json:"senderAddress"`
+	FeeRecipientAddress   common.Address `json:"feeRecipientAddress"`
+	ExpirationTimeSeconds *big.Int       `json:"expirationTimeSeconds"`
+	Salt                  *big.Int       `json:"salt"`
+	Signature             []byte         `json:"signature"`
+}
```

Comment:
Good catch. We need both `Order` and `NewOrder` in the GraphQL schema itself because GraphQL requires that any type used in mutations must be an `input` type. That also carries over to the generated server code. However, we don't actually need two distinct types in the client code.